app/src/common/shared/org/mozilla/vrbrowser/browser/engine/GeckoEngineSession.java

```diff
@@ -0,0 +1,112 @@
+package org.mozilla.vrbrowser.browser.engine;
+
+import androidx.annotation.NonNull;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.mozilla.geckoview.GeckoSession;
+import org.mozilla.vrbrowser.browser.engine.Session;
+
+import mozilla.components.concept.engine.EngineSession;
+import mozilla.components.concept.engine.EngineSessionState;
+import mozilla.components.concept.engine.Settings;
+
+public class GeckoEngineSession extends EngineSession {
+
+    private Session mSession;
+
+    public GeckoEngineSession(@NonNull Session session) {
+        mSession = session;
+    }
+
+    public GeckoSession getGeckoSession() {
+        return mSession.getGeckoSession();
+    }
+
+    @NotNull
+    @Override
+    public Settings getSettings() {
+        return null;
+    }
+
```

Comment:
This one too, returning `null` from `@NotNull`

Reply:
I've added support for Settings although we are not using it ATM. Our bridge is merely for handling the session state in the web extensions features.